From 3cc44d351aaf5c7673dd72c70432cce6bcad353a Mon Sep 17 00:00:00 2001
From: Beleswar Padhi <b-padhi@ti.com>
Date: Thu, 19 Jun 2025 16:31:04 +0530
Subject: [PATCH 2/6] TDA54: irqchip: irq-ti-sci-intr: Add support to map
 multiple mbox IRQs

With multiple remote processors being enabled, newer mailbox cluster
have been enabled for IPC. Map the new input mailbox irq
(MAILBOX0_CLUSTER1) to a valid GIC line.

Signed-off-by: Beleswar Padhi <b-padhi@ti.com>
---
 drivers/irqchip/irq-ti-sci-intr.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/irqchip/irq-ti-sci-intr.c b/drivers/irqchip/irq-ti-sci-intr.c
index 1ca535727889..ff86019016f5 100644
--- a/drivers/irqchip/irq-ti-sci-intr.c
+++ b/drivers/irqchip/irq-ti-sci-intr.c
@@ -23,7 +23,8 @@
 #define INTR_OUT_CNT 164
 #define INTR_BCDMA_START 328
 #define INTR_BCDMA_END 301
-#define MAILBOX_INTR 381
+#define MAILBOX0_CLUSTER0_INTR 381
+#define MAILBOX0_CLUSTER1_INTR 377
 
 /**
  * struct to distinguish between interrupt router which does not use ti-sci for
@@ -292,8 +293,11 @@ static void update_navss_map(u16 *map)
 		map[i] = gic_cnt++;
 		irq_routed_count++;
 	}
-	if (irq_routed_count < INTR_OUT_CNT)
-		map[MAILBOX_INTR] = gic_cnt;
+	if (irq_routed_count++ < INTR_OUT_CNT)
+		map[MAILBOX0_CLUSTER0_INTR] = gic_cnt++;
+	if (irq_routed_count++ < INTR_OUT_CNT)
+		map[MAILBOX0_CLUSTER1_INTR] = gic_cnt++;
+
 }
 
 static const struct irq_domain_ops ti_sci_intr_irq_domain_ops = {
-- 
2.34.1

